### ISSUE TITLE
Question about entropy coding 

### ISSUE THREAD

**Issue by net-F** [3dpcc/DeepPCC#2]
Amazing work ！

May I ask that did you use compressai to implement the Hyper Prior and Autoregressive entropy model, as well as corresponding coding/compressing ?

**Comment by 3dpcc** [comment]
Thank you for your question!

Instead of using the Compressai library, we replicated Hyperprior and Autoregressive designs for 3D point cloud compression with MinkowskiEngine, where most 3D Convolution is replaced by Sparse Convolution and Mask Sparse Convolution. So, the framework does not use Compressai.

**Comment by net-F** [comment]
Thank you for your answer!

What about the entropy coding ? Did you reuse the entropy coding of Compressai library or torchac library or implement it by yourself ?

**Comment by net-F** [comment]
Is there a significant difference in performance between using Gaussian distribution and using Laplace distribution ？

**Comment by 3dpcc** [comment]
We use torchac library to coding the latent feature value. The difference in performance between Gaussian distribution and Laplace distribution  is small. I feel that using Laplace distribution is more convergent in entropymodel training.

**Comment by net-F** [comment]
Thanks for your answer!

Is there a significant difference in performance between using and not using autoregressive models ？ Does main difference exist in low rates or whole rd curve?

**Comment by net-F** [comment]
I notice that your learning rate will quickly decay to 1e-5 at first ten epochs,  and the traning will be at 1e-5 in most epochs. However, in my own toy implementation the loss decreasing is at relatively high learning rate training, and need much more epochs to achieve a normal performance. How did you get a such good performance in such few epochs and low learning rates.

**Comment by 3dpcc** [comment]
Autoregressive will bring a BD-Rate Gain of 10% compared with only using hyperprior model. But the time complexity is very high. Therefore, there is a trade-off between performance and complexity. 

Our training set is close to 10,000 sequences one epoch, so we set the learning rate drops every two epoch. When network training, we will also watch the change of loss, and also adjust the learning rate in a certain epoch to continue training.
